epoch_failover_test.go

```diff
@@ -61,30 +61,22 @@ func TestEpochLeaderFailover(t *testing.T) {
 	// Then, don't do anything and wait for our node
 	// to start complaining about a block not being notarized
 
-	for _, round := range []uint64{0, 1, 2} {
+	for round := uint64(0); round < 3; round++ {
 		notarizeAndFinalizeRound(t, nodes, round, round, e, bb, quorum, storage, false)
 	}
 
 	bb.blockShouldBeBuilt <- struct{}{}
 
 	waitForEvent(t, start, e, timeoutDetected)
-
-	lastBlock, _, ok := storage.Retrieve(storage.Height() - 1)
-	require.True(t, ok)
-
-	prev := lastBlock.BlockHeader().Digest
-
-	md := ProtocolMetadata{
-		Round: 3,
-		Seq:   2,
-		Prev:  prev,
-	}
+	
+	emptyBlockMd := e.Metadata()
```

Comment:
I would rather explicitly craft the block manually and not rely on the metadata because I want whoever reads the test to understand what's going on with the sequences and rounds, and also I don't want to rely on the correctness of the Metadata(), as this is essentially test input.